tests/test_bdd.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+from pytest_bdd import scenario, given, when, then
+import pytest
+import markdown
+
+import journal
+
+
+# Homepage Feature
+@scenario('features/homepage.feature',
+          'Homepage lists of entries')
+def test_home_listing_as_anon():
+    pass
+
+
+@given('I am an anonymous user')
+def an_anonymous_user(app):
+    pass
+
+
+@given('I have three entries')
+def create_entries(db_session):
+    title_template = "Title {}"
+    text_template = "Entry Text {}"
+    for x in range(3):
+        journal.Entry.write(
+            title=title_template.format(x),
+            text=text_template.format(x),
+            session=db_session
+        )
+        db_session.flush()
+
+
+@when('I go to homepage')
+def homepage(homepage):
+    pass
+
+
+@then('I see three entries')
+def check_entry_list(homepage):
+    html = homepage.html
+    entries = html.find_all('article', class_='entry')
+    assert len(entries) == 3
+
+
+# Detail Feature
+@scenario('features/detail.feature',
+          'Display a detailed entry linked from index page')
+def test_detail_listing():
+    pass
+
+
+@given('I have a journal entry')
+def create_one_entry(db_session):
+    journal.Entry.write(
+        title='Title',
+        text='text',
+        session=db_session
+    )
+    db_session.flush()
+
+
+@when('I go to the detail page from index page')
+def visit_detail_page(app):
+    pass
+
+
+@then('I see the entry')
+def check_detail_entry(app):
+    response = app.get('/detail/4')
```

Comment:
watch out for hard-coding the ID from an entry into this step.  If this step depended on the `create_one_entry` fixture above (line 55), you'd be able to return the entry you created in that fixture and here it would be available.  You could then check the `id` of that entry to build the correct url:

```python
def create_one_entry(db_session):
    entry = journal.Entry.write(...)
    db_session.flush()
    return entry

def check_detail_entry(app, create_one_entry)
    entry_id = create_one_entry.id
    url = '/detail/{}'.format(entry_id)
    response = app.get(url)
    ...
```